### PR TITLE
dmd: build using gcc11Stdenv

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

The DMD test suite has been failing since we updated to gcc12, as
reported by @trofi:

  https://github.com/NixOS/nixpkgs/pull/206907#issuecomment-1527034123

Let's use gcc11Stdenv for now.

https://github.com/NixOS/nixpkgs/issues/230712

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build dmd dmd.passthru.tests

**Review by wegank — DISMISSED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6836,6 +6836,8 @@ with pkgs;
 
   dmd = callPackage ../development/compilers/dmd {
     inherit (darwin.apple_sdk.frameworks) Foundation;
+    # https://github.com/NixOS/nixpkgs/pull/206907#issuecomment-1527034123
+    stdenv = gcc11Stdenv;
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:6840]:
```suggestion
    stdenv = if stdenv.isLinux then gcc11Stdenv else stdenv;
```

**Review by trofi — APPROVED** [review]
LGTM, i agree we should not override stdenv at least for darwin.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (amjoseph-nixpkgs) [reply pkgs/top-level/all-packages.nix:6840]: For some reason github is not letting me apply your suggestion ("This diff has recently been updated.  Refresh and try again" -- but I refresh and still get the error).

I will apply your suggestion manually.

**Comment by amjoseph-nixpkgs** [comment]
Squashed (into two commits instead of three), rebased.

**Comment by amjoseph-nixpkgs** [comment]
Waiting for ofborg-eval to finish, then I will merge this.

**Pull request merged**